tests/unit/remote/test_s3.py

```diff
@@ -0,0 +1,89 @@
+import boto3
+import os
+import pytest
+from moto import mock_s3
+
+from dvc.remote.s3 import RemoteS3
+
+
+@pytest.fixture
+def aws_credentials():
+    """Mocked AWS Credentials for moto."""
+    os.environ["AWS_ACCESS_KEY_ID"] = "testing"
+    os.environ["AWS_SECRET_ACCESS_KEY"] = "testing"
+    os.environ["AWS_SECURITY_TOKEN"] = "testing"
+    os.environ["AWS_SESSION_TOKEN"] = "testing"
+
+
+@pytest.fixture
+def s3(aws_credentials):
+    """Returns a connection to a bucket with the following file structure:
+
+        bucket
+        ├── data
+        │  ├── alice
+        │  ├── alpha
+        │  └── subdir
+        │     ├── 1
+        │     ├── 2
+        │     └── 3
+        ├── empty_dir
+        ├── empty_file
+        └── foo
+    """
+    with mock_s3():
+        s3 = boto3.client("s3", region_name="us-east-1")
+        s3.create_bucket(Bucket="bucket")
+
+        s3.put_object(Bucket="bucket", Key="empty_dir/")
+        s3.put_object(Bucket="bucket", Key="empty_file", Body=b"")
+        s3.put_object(Bucket="bucket", Key="foo", Body=b"foo")
+        s3.put_object(Bucket="bucket", Key="data/alice", Body=b"alice")
+        s3.put_object(Bucket="bucket", Key="data/alpha", Body=b"alpha")
+        s3.put_object(Bucket="bucket", Key="data/subdir/1", Body=b"1")
+        s3.put_object(Bucket="bucket", Key="data/subdir/2", Body=b"2")
+        s3.put_object(Bucket="bucket", Key="data/subdir/3", Body=b"3")
+
+        yield s3
+
+
+@pytest.fixture
+def remote():
+    """Returns RemoteS3 instance to work with the `s3` fixture."""
+    yield RemoteS3(None, {"url": "s3://bucket", "region": "us-east-1"})
+
+
+def test_isdir(s3, remote):
+    assert remote.isdir(remote.path_info / "data")
```

Comment:
you could totally use `parametrize` here instead of copypasting. :slightly_smiling_face:

Reply:
Yes! Forgot about `parametrize`  :ok_hand:

Reply:
Interesting, `parametrize` is 1 and a half second slower :sweat:

Reply:
@mroutis this is because all the fixtures are rebuilt every time. Use for loop inside single test instead. Moreover, I would reuse the same list for fixture and tests as far as that is possible.